level WARN still fails the build

```
n: CLASS_NOT_FOUND, target class location not found, isReachable: true
  MethodSymbolReference{sourceClassName=net.bytebuddy.agent.VirtualMachine$ForHotSpot$OnUnix, targetClassName=org.newsclub.net.unix.AFUNIXSocket, methodName=close, interfaceMethod=false, descriptor=()V}, reason: CLASS_NOT_FOUND, target class location not found, isReachable: true


[WARNING] Rule 0: com.google.cloud.tools.dependencies.enforcer.LinkageCheckerRule failed with message:
Failed while checking class path. See above error report.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10:21 min
[INFO] Finished at: 2019-04-04T12:59:23-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M1:enforce (linkage) on project appengine-plugins-core: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java
+++ b/enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java
@@ -155,10 +155,11 @@ public class LinkageCheckerRule extends AbstractNonCacheableEnforcerRule {
             logger.warn(message);
           } else {
             logger.error(message);
+            throw new EnforcerRuleException(
+                "Failed while checking class path. See above error report.");
           }
-          throw new EnforcerRuleException(
-              "Failed while checking class path. See above error report.");
         } else {
+          // arguably shouldn't log anything on success
           logger.info("No " + foundError + " found");
         }
       } catch (IOException ex) {

--- a/enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java
+++ b/enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java
@@ -30,6 +30,7 @@ import java.util.Arrays;
 import java.util.Objects;
 import java.util.stream.Collectors;
 import org.apache.maven.artifact.handler.DefaultArtifactHandler;
+import org.apache.maven.enforcer.rule.api.EnforcerLevel;
 import org.apache.maven.enforcer.rule.api.EnforcerRuleException;
 import org.apache.maven.enforcer.rule.api.EnforcerRuleHelper;
 import org.apache.maven.execution.MavenSession;
@@ -179,6 +180,20 @@ public class LinkageCheckerRuleTest {
       Assert.assertEquals(
           "Failed while checking class path. See above error report.", ex.getMessage());
     }
+  }
+
+  @Test
+  public void testExecute_shouldPassForBadProject_levelWarn()
+      throws RepositoryException, EnforcerRuleException {
+    // This pair of artifacts contains linkage errors on grpc-core's use of Verify. Because
+    // grpc-core is included in entry point jars, the errors are reachable.
+    setupMockDependencyResolution(
+        "com.google.api-client:google-api-client:1.27.0", "io.grpc:grpc-core:1.17.1");
+    rule.setReportOnlyReachable(true);
+    rule.setLevel(EnforcerLevel.WARN);
+    rule.execute(mockRuleHelper);
+    verify(mockLog)
+        .warn(ArgumentMatchers.startsWith("Linkage Checker rule found reachable errors."));
   }
 
   @Test

